### ISSUE TITLE
[BUG] Fleet-Agent not being deployed

### ISSUE THREAD

**Issue by strowi** [rancher/rancher#47016]
**Rancher Server Setup**
- Rancher version: 2.9.1
- Installation option (Docker install/Helm Chart): k3s
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): 1.28.8 and up
- Proxy/Cert Details: LetsEncrypt

**Information about the Cluster**
- Kubernetes version: 1.28.8
- Cluster Type (Local/Downstream): k3s
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider): imported

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom)
  - If custom, define the set of permissions: Admin

**Describe the bug**

Fleet is not deployed on downstream-cluster

**To Reproduce**

- Setup Rancher 2.9.1
- import k3s-cluster into rancher
 
**Result**
- cluster-agent is deployed
- fleet-agent is not deployed, Rancher--> CD -> Clusters show the error:
` {"Code":{"Code":"Forbidden","Status":403},"Message":"clusters.management.cattle.io \"c-m-95v7sk4s\" is forbidden: User \"u-cfuukbmkzo\" cannot get resource \"clusters\" in API group \"management.cattle.io\" at the cluster scope","Cause":null,"FieldName":""}`

**Expected Result**

Fleet being deployed on downstream cluster

**Screenshots**

![image](https://github.com/user-attachments/assets/b465b57d-0281-4c4d-8d8f-9c0cbe46745e)

**Additional context**

Logs from the cluster-agent:
```bash

kn cattle-system logs cattle-cluster-agent-55b5bbd94-qcm5m|grep -i error

time="2024-09-10T11:54:04Z" level=error msg="unable to read CA file from /etc/kubernetes/ssl/certs/serverca: open /etc/kubernetes/ssl/certs/serverca: no such file or directory"
time="2024-09-10T11:54:13Z" level=error msg="namespaceHandler: Sync: error adding project id label to namespace err=Operation cannot be fulfilled on namespaces \"kube-public\": the object has been modified; please apply your changes to the latest version and try again"
time="2024-09-10T11:58:07Z" level=error msg="Unknown error: open /var/lib/rancher-data/local-catalogs/v2/rancher-charts/4b40cac650031b74776e87c1a726b0484d0877c3ec137da0872547ff9b73a721/assets/logos/rancher-monitoring.png: no such file or directory"
time="2024-09-10T11:58:07Z" level=error msg="Unknown error: open /var/lib/rancher-data/local-catalogs/v2/rancher-charts/4b40cac650031b74776e87c1a726b0484d0877c3ec137da0872547ff9b73a721/assets/logos/rancher-logging.svg: no such file or directory"
time="2024-09-10T11:58:39Z" level=error msg="Error during subscribe websocket: close sent"

```

**Comment by strowi** [comment]
Update: Editing the fleet-deployment ond the downstream-cluster and changing `CATTLE_FEATURES: ...,fleet=false,...` to `true` doesdeploy fleet,  But it doesn't change the Rancher-Error or deploy anything else.

**Comment by strowi** [comment]
Update: Ok after some more hours of debugging:

As far as i can tell the Clusterrole was/is missing the permission to get the cluster-info on the rancher side.

Using the user-id from the error-message i was able to hunt down the clusterrolebinding to the clusterrole, which looked like this: 

```
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  creationTimestamp: "2024-09-10T13:55:43Z"
  finalizers:
  - wrangler.cattle.io/auth-prov-v2-crole
  labels:
    cattle.io/creator: norman
  name: u-brr5d5s2s7-view
  ownerReferences:
  - apiVersion: management.cattle.io/v3
    kind: User
    name: u-brr5d5s2s7
    uid: ed05dea8-6cea-4942-92da-79106467b3bf
  resourceVersion: "305943324"
  uid: 1017c453-4f05-4131-9612-382b52561bea
rules:
- apiGroups:
  - management.cattle.io
  resourceNames:
  - u-brr5d5s2s7
  resources:
  - users
  verbs:
  - get
 ```

Adding the following permission and force-updating the cluster made it work again:

```
- apiGroups:
  - management.cattle.io
  resourceNames:
  - c-jt58n
  resources:
  - clusters
  verbs:
  - get
 ```

regards,
strowi

**Comment by strowi** [comment]
This seems to have solved itself by either re-deploying 2.9.2 or upgrading to 2.9.3.
